### ISSUE TITLE
Should handle processors returned from plugins

### ISSUE THREAD

**Issue by ben-eb** [postcss/postcss#402]
PostCSS should be able to handle plugins that return processors, such as cssnext & cssnano. This test fails:

```js
test('can be used as a postcss plugin without parens', function (t) {
    var css = 'h1 { color: #ffffff }';
    var min = 'h1{color:#fff}';

    var out = postcss(nano).process(css).css;

    t.plan(1);
    t.equal(out, min, specName('beConsumedByPostCSS'));
});
```

```
TAP version 13
# can be used as a postcss plugin without parens
/Users/beneb/projects/cssnano/node_modules/postcss/lib/lazy-result.js:196
            throw error;
                  ^
TypeError: plugin is not a function
    at LazyResult.run (/Users/beneb/projects/cssnano/node_modules/postcss/lib/lazy-result.js:193:24)
    at LazyResult.sync (/Users/beneb/projects/cssnano/node_modules/postcss/lib/lazy-result.js:179:32)
    at LazyResult.stringify (/Users/beneb/projects/cssnano/node_modules/postcss/lib/lazy-result.js:210:14)
    at LazyResult.get (/Users/beneb/projects/cssnano/node_modules/postcss/lib/lazy-result.js:232:25)
    at Test.<anonymous> (/Users/beneb/projects/cssnano/tests/api.js:12:41)
    at Test.bound [as _cb] (/Users/beneb/projects/cssnano/node_modules/tape/lib/test.js:62:32)
    at Test.run (/Users/beneb/projects/cssnano/node_modules/tape/lib/test.js:75:10)
    at Test.bound [as run] (/Users/beneb/projects/cssnano/node_modules/tape/lib/test.js:62:32)
    at Immediate.next [as _onImmediate] (/Users/beneb/projects/cssnano/node_modules/tape/lib/results.js:66:15)
    at processImmediate [as _immediateCallback] (timers.js:371:17)
```

Whereas this passes:

```js
test('can be used as a postcss plugin', function (t) {
    var css = 'h1 { color: #ffffff }';
    var min = 'h1{color:#fff}';

    var out = postcss(nano()).process(css).css;

    t.plan(1);
    t.equal(out, min, specName('beConsumedByPostCSS'));
});
```

**Comment by ai** [comment]
Fixed f77611d

**Comment by ai** [comment]
Released in 4.1.14

**Comment by ben-eb** [comment]
Awesome, thanks! :grinning:
